app/modules/power/migrations/004_power_schedule_text.php

```diff
@@ -0,0 +1,73 @@
+
+                    // Copy over data from old table to new table
+                    $sql = 'INSERT INTO power(id, serial_number, manufacture_date, design_capacity, max_capacity, max_percent, current_capacity, current_percent, cycle_count, temperature, condition, timestamp, sleep_prevented_by, hibernatefile, schedule, adapter_id, family_code, adapter_serial_number, combined_sys_load, user_sys_load, thermal_level, battery_level, ups_name, active_profile, ups_charging_status, externalconnected, cellvoltage, manufacturer, batteryserialnumber, fullycharged, ischarging, standbydelay, standby, womp, halfdim, gpuswitch, sms, networkoversleep, disksleep, sleep, autopoweroffdelay, hibernatemode, autopoweroff, ttyskeepawake, displaysleep, acwake, lidwake, sleep_on_power_button, autorestart, destroyfvkeyonstandby, powernap, haltlevel, haltafter, haltremain, lessbright, sleep_count, dark_wake_count, user_wake_count, wattage, backgroundtask, applepushservicetask, userisactive, preventuseridledisplaysleep, preventsystemsleep, externalmedia, preventuseridlesystemsleep, networkclientactive, cpu_scheduler_limit, cpu_available_cpus, cpu_speed_limit, ups_percent, timeremaining, instanttimetoempty, voltage, permanentfailurestatus, packreserve, avgtimetofull, amperage, designcyclecount, avgtimetoempty) SELECT id, serial_number, manufacture_date, design_capacity, max_capacity, max_percent, current_capacity, current_percent, cycle_count, temperature, condition, timestamp, sleep_prevented_by, hibernatefile, schedule, adapter_id, family_code, adapter_serial_number, combined_sys_load, user_sys_load, thermal_level, battery_level, ups_name, active_profile, ups_charging_status, externalconnected, cellvoltage, manufacturer, batteryserialnumber, fullycharged, ischarging, standbydelay, standby, womp, halfdim, gpuswitch, sms, networkoversleep, disksleep, sleep, autopoweroffdelay, hibernatemode, autopoweroff, ttyskeepawake, displaysleep, acwake, lidwake, sleep_on_power_button, autorestart, destroyfvkeyonstandby, powernap, haltlevel, haltafter, haltremain, lessbright, sleep_count, dark_wake_count, user_wake_count, wattage, backgroundtask, applepushservicetask, userisactive, preventuseridledisplaysleep, preventsystemsleep, externalmedia, preventuseridlesystemsleep, networkclientactive, cpu_scheduler_limit, cpu_available_cpus, cpu_speed_limit, ups_percent, timeremaining, instanttimetoempty, voltage, permanentfailurestatus, packreserve, avgtimetofull, amperage, designcyclecount, avgtimetoempty FROM power_tmp'
+                    $this->exec($sql);
+                
+                    // Delete old table
+                    $sql = 'DROP TABLE power_tmp';
+                    $this->exec($sql);
+                
+                // Get database handle
+                $dbh = $this->getdbh();
+
+                $dbh->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+                // Wrap in transaction
+                $dbh->beginTransaction();
```

Comment:
Please move beginTransaction to the start of the sqlite section. Transactions make the migration atomic.